File: src/main/java/com/rokucraft/RokuBot/entities/DiscordInvite.java
package com.rokucraft.RokuBot.entities;

import com.rokucraft.RokuBot.Settings;
import org.checkerframework.checker.nullness.qual.Nullable;
import org.spongepowered.configurate.objectmapping.ConfigSerializable;

@ConfigSerializable
public class DiscordInvite extends AbstractEntity{
    private String inviteCode;

    public DiscordInvite() {}

    public DiscordInvite(String name, String[] aliases, boolean staffOnly, String inviteCode) {
        this.name = name;
        this.aliases = aliases;
        this.staffOnly = staffOnly;
        this.inviteCode = inviteCode;
        Settings.discordInviteList.add(this);
    }

    @Nullable
    public String getInviteUrl() {
        if (inviteCode != null && !inviteCode.isEmpty()) {
            return "https://discord.gg/" + inviteCode;
        } else return null;
    }

    @Nullable
    public static DiscordInvite find(String name) {
        return (DiscordInvite) find(name, Settings.discordInviteList);
    }
}

File: src/main/kotlin/entities/DiscordInvite.kt
package com.rokucraft.rokubot.entities

import org.spongepowered.configurate.objectmapping.ConfigSerializable
import org.spongepowered.configurate.objectmapping.meta.Required

@ConfigSerializable
data class DiscordInvite(
    @Required val name: String,
    @Required val inviteCode: String,
) {
    fun inviteUrl(): String = "https://discord.gg/$inviteCode"
}
